clients/MONGODB/mongodb.js

```diff
@@ -9,27 +8,21 @@ module.exports = {
     type: ''
   },
   transform: (binding) => {
-    let userPassword = '';
-    if (binding.connectionOptions &&
-        binding.connectionOptions.auth &&
-        binding.connectionOptions.auth.username) {
-      const encodedUser =
-        encodeURIComponent(binding.connectionOptions.auth.username);
-      let encodedPassword = '';
-      if (binding.connectionOptions.auth.password) {
-        encodedPassword =
-          encodeURIComponent(binding.connectionOptions.auth.password);
-      }
-      userPassword = `${encodedUser}:${encodedPassword}@`;
-    }
-    if (binding.srv === 'true') {
-      binding.url = `mongodb+srv://${userPassword}${binding.host}`;
-    } else {
-      let port = '';
-      if (binding.port) {
-        port = `:${binding.port}`;
-      }
-      binding.url = `mongodb://${userPassword}${binding.host}${port}`;
-    }
+    const encodedUser = encodeURIComponent(
+      binding.connectionOptions.auth.username
+    );
+
+    const encodedPassword = binding.connectionOptions.auth.password
+      ? encodeURIComponent(binding.connectionOptions.auth.password)
+      : '';
+
+    const srv = binding.srv === 'true';
+
+    binding.url = [
+      'mongodb',
+      srv ? '+srv' : '',
```

Comment:
I don't think this woud support not having a userid, which I think might be possible and the old code supported?

Reply:
the userid is the encodeduser variable, which is part of the mongo uri

Reply:
My point is that from https://www.mongodb.com/docs/manual/reference/connection-string/

the username:password is optional, so it is possible to connect without it.  The old code handed that but I don't think the modified code does.

Reply:
We can address this comment in a follow on PR if necessary.